src/gamestate_handler.h

```diff
@@ -0,0 +1,18 @@
+#pragma once 
+#include <map>
+
+#include <SDL.h>
+#include "I_gamestate.h"
+
+class GameStateHandler {
+public:
+	GameStateHandler(std::map<gamestates::GameStateName, I_GameState*>* gamestates);
+	void set_renderer(SDL_Renderer* renderer);
```

Comment:
So AFAIK, we only need one renderer in this whole program. Basically, your main right now is going to be one "state", and game states take a renderer. I know in pygame we passed one surface around to ensure one window, I was thinking the renderer would be the equivalent for SDL 2.0. I wasn't sure, so let me know how it should work. 

But either way, I would highly suggest looking at GameState.h for possibly a bit more clarification. The test should help too.